common/djangoapps/third_party_auth/models.py

```diff
@@ -361,24 +347,9 @@ class SAMLProviderConfig(ProviderConfig):
             "URL to this provider's XML metadata. Should be an HTTPS URL. "
             "Example: https://www.testshib.org/metadata/testshib-providers.xml"
         ))
-    attr_user_permanent_id = models.CharField(
-        max_length=128, blank=True, verbose_name="User ID Attribute",
-        help_text="URN of the SAML attribute that we can use as a unique, persistent user ID. Leave blank for default.")
-    attr_full_name = models.CharField(
-        max_length=128, blank=True, verbose_name="Full Name Attribute",
-        help_text="URN of SAML attribute containing the user's full name. Leave blank for default.")
-    attr_first_name = models.CharField(
-        max_length=128, blank=True, verbose_name="First Name Attribute",
-        help_text="URN of SAML attribute containing the user's first name. Leave blank for default.")
-    attr_last_name = models.CharField(
-        max_length=128, blank=True, verbose_name="Last Name Attribute",
-        help_text="URN of SAML attribute containing the user's last name. Leave blank for default.")
-    attr_username = models.CharField(
-        max_length=128, blank=True, verbose_name="Username Hint Attribute",
-        help_text="URN of SAML attribute to use as a suggested username for this user. Leave blank for default.")
-    attr_email = models.CharField(
-        max_length=128, blank=True, verbose_name="Email Attribute",
-        help_text="URN of SAML attribute containing the user's email address[es]. Leave blank for default.")
+    attributes = models.TextField(
```

Comment:
Any reason not to use a `JSONField`? (do a git grep on edx-platform to see usage)